src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -215,17 +215,34 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Default(byte* pBuffer, n
             goto Finish;
         }
 
-        private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuint bufferLength)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ulong GetMaskOfMostSignificantBits(Vector128<byte> value)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian);
+
+            // extractedBits[i] = (value[i] & 0x80 == 0x80) & (1 << (12 * (i % 2)))
+            Vector128<byte> s_mostSignficantBitMask = Vector128.Create((byte)0x80);
+            Vector128<byte> s_bitmask = Vector128.Create((ushort)0x1001).AsByte();
+            Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, s_mostSignficantBitMask), s_mostSignficantBitMask);
```

Comment:
I think functionality of
```c#
Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, s_mostSignficantBitMask), s_mostSignficantBitMask);
```
can be achieved with
```c#
Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareTest(value, s_mostSignficantBitMask);
```
since each element of s_mostSignficantBitMask has only one bit set.

Reply:
Actually, there is a better approach
```c#
Vector128<byte> mostSignificantBitIsSet = AdvSimd.ShiftRightArithmetic(value.AsSByte(), 7).AsByte();
```
eliminating need in `s_mostSignficantBitMask`.